Fix beastify code

aloopuri opened the pull request:
#502 introduced an error where if you click on the background of the popup or if you added anything else to the popup page and clicked on it would try and run the beastify function, and this would cause the screen to go blank.

This issue was this else statement which accepted any and all clicks on the popup page.

https://github.com/mdn/webextensions-examples/blob/6d8f07243a23c6b402cc33c7ac926155f524f3ce/beastify/popup/choose_beast.js#L68-L78

estelle commented on beastify/popup/choose_beast.html:
```diff
@@ -8,10 +8,10 @@
-    <button>Turtle</button>
-    <button>Snake</button>
-    <button type="reset">Reset</button>
+    <button class="button beast">Frog</button>
```
this shouldn't be necessary. Just style based on `<button>` and `[type="reset"]`

estelle commented on beastify/popup/choose_beast.css:
```diff
@@ -6,25 +6,28 @@ html, body {
 }
 
-button[type="reset"] {
+.beast {
```
is there a reason we to add classes? Is there something about extensions that make type selectors and attribute selectors not work??

estelle replied:
Or is this just reverting 502?

estelle reviewed (commented):
This PR adds several class names that don't seem necessary. Are they?

aloopuri commented:
I added a new fix. I reverted back to using types again as you suggested and added a type to the other buttons too, which fixes the issue I originally mentioned (i.e. the else statement allowing all clicks and showing a blank page when the background is clicked). Also it's good practice as mentioned [here](https://dev.to/clairecodes/why-its-important-to-give-your-html-button-a-type-58k9)

I thought of using classes again because of the event handler for the entire page. If someone was to extend this project and add more buttons with different functionality (such as transforming the images), we would need another way of identifying what those buttons are, as identifying them based on only their type attribute wouldn't be possible, would it? But I guess that may be beyond the scope of this project anyway.

Rob--W commented on beastify/popup/choose_beast.html:
```diff
@@ -8,9 +8,9 @@
-    <button>Frog</button>
-    <button>Turtle</button>
-    <button>Snake</button>
+    <button type="button">Frog</button>
```
`type="button"` is not explicitly needed. By default it's `type="submit"`.

Rob--W commented on beastify/popup/choose_beast.js:
```diff
@@ -65,13 +65,13 @@ function listenForClicks() {
      * then call "beastify()" or "reset()" as appropriate.
      */
-    if (e.target.type === "reset") {
+    if (e.target.type === "button") {
```
Instead of being indirect here, let's just prepend:

```lang=javascript
if (e.target.tagName !== "BUTTON" || !e.target.closest("#popup-content")) {
  // Ignore when click is not on a button within <div id="popup-content">.
  return;
}
```

aloopuri commented on beastify/popup/choose_beast.js:
```diff
@@ -65,7 +65,10 @@ function listenForClicks() {
+    if (e.target.tagName !== "BUTTON" || !e.target.closest("#popup-content")) {
+      // Ignore when click is not on a button within <div id="popup-content">.
+      return;
+    } else if (e.target.type === "reset") {
```
Oh ok I'll fix it now

Rob--W reviewed (approved):
Thanks for the fix!
